### PR TITLE
Add UniqueAlphaNumericScrambler

### PR DESCRIPTION
**Pull request by RouganStriker**
Related to #324

### CODE REVIEW

File: processors.go [hunk 1]
```diff
@@ -121,29 +184,30 @@ var UUIDMap = safeUUIDMap{
...
-		"FakeUsername":          ProcessorUserName,
-		"FakeZip":               ProcessorZip,
-		"Identity":              ProcessorIdentity, // Default: Does not modify field
-		"RandomBoolean":         ProcessorRandomBoolean,
-		"RandomDate":            ProcessorRandomDate,
-		"RandomDigits":          ProcessorRandomDigits,
-		"RandomUUID":            ProcessorRandomUUID,
-		"ScrubString":           ProcessorScrubString,
+		"AlphaNumericScrambler":        ProcessorAlphaNumericScrambler,
+		"UniqueAlphaNumericScrambler":  ProcessorUniqueAlphaNumericScrambler,
```
Comment (junkert) [inline processors.go:188]: Just a nit, but is it ok if we keep this list separated alphabetically?

File: processors_test.go [hunk 2]
```diff
@@ -79,6 +81,142 @@ func TestProcessorAlphaNumericScrambler(t *testing.T) {
...
+
+	require.NotNil(t, err)
+	require.Equal(t, output, "")
+}
+
+func TestProcessorUniqueAlphaNumericScramblerEnsuresUniquenessPerColumn(t *testing.T) {
+	rand.Seed(5336)
+
+	// Use a fixed seed to pre-generate the scrambled strings
+	preGeneratedUniqueStrings := []string{
```
Comment (junkert) [inline processors_test.go:174]: Maybe move this to a constant global and call if from within the two functions? They look identical from here, but can't quite tell. If they are different ignore me :)

File: processors_test.go [hunk 3]
```diff
@@ -79,6 +81,142 @@ func TestProcessorAlphaNumericScrambler(t *testing.T) {
...
+		require.Equal(t, len(outputEscaped), len(input))
+		require.True(t, strings.HasPrefix(outputEscaped, completeEscapeSequences[i]))
+		require.False(t, strings.HasSuffix(outputEscaped, suffix))
+	}
+}
+
+func TestProcessorUniqueAlphaNumericScramblerEnsuresUniqueness(t *testing.T) {
+	rand.Seed(5336)
+
+	// Use a fixed seed to pre-generate the scrambled strings
```
Comment (junkert) [inline processors_test.go:129]: I think this comment is for line 127

File: processors_test.go [hunk 4]
```diff
@@ -79,6 +81,142 @@ func TestProcessorAlphaNumericScrambler(t *testing.T) {
...
+	output, err := ProcessorUniqueAlphaNumericScrambler(&cMap, "This is my unscrambled text")
+
+	require.NotNil(t, err)
+	require.Equal(t, output, "")
+}
+
+func TestProcessorUniqueAlphaNumericScramblerEnsuresUniquenessPerColumn(t *testing.T) {
+	rand.Seed(5336)
+
+	// Use a fixed seed to pre-generate the scrambled strings
```
Comment (junkert) [inline processors_test.go:173]: same here for the comment

**Review by junkert — COMMENTED** [review]
@RouganStriker thank you for adding this! Just a couple nits then we can merge.

File: processors.go (hunk 1 above)
Comment (RouganStriker) [reply processors.go]: Alpha-ordered this and updated the README as well.

File: processors_test.go (hunk 2 above)
Comment (RouganStriker) [reply processors_test.go]: Yup they are identical, I moved it up to a global variable

File: processors_test.go (hunk 3 above)
Comment (RouganStriker) [reply processors_test.go]: Moved comment

File: processors_test.go (hunk 4 above)
Comment (RouganStriker) [reply processors_test.go]: Moved comment

**Comment by RouganStriker** [comment]
Thanks for the feedback! I've update the PR with the requested changes

**Review by junkert — APPROVED** [review]
Thanks again!

**Pull request merged**